tests/cupy_tests/random_tests/test_distributions.py

```diff
@@ -1,9 +1,32 @@
 import unittest
 
+import numpy
+
+import cupy
+from cupy.random import distributions
 from cupy import testing
 
 
+@testing.parameterize(*testing.product({
+    'shape': [(3, 2)],
+    'loc_shape': [(), (3, 2)],
+    'scale_shape': [(), (3, 2)]
+})
+)
 @testing.gpu
 class TestDistributions(unittest.TestCase):
 
     _multiprocess_can_split_ = True
+
+    def check_distribution(self, dist_func, dtype):
+        loc = cupy.ones(self.loc_shape)
+        scale = cupy.ones(self.scale_shape)
+        out = dist_func(loc, scale, self.shape, dtype)
+        self.assertEqual(self.shape, out.shape)
+        self.assertEqual(out.dtype, dtype)
+
+    def test_normal_float32(self):
+        self.check_distribution(distributions.normal, numpy.float32)
```

Comment:
How about using [`for_float_dtypes`](http://docs.chainer.org/en/stable/cupy-reference/testing.html#cupy.testing.for_float_dtypes) (or other variants) to parameterize dtype?

An example usage is [here](https://github.com/pfnet/chainer/blob/8edf6b601ce2669d2f4c812d1bc8a62e78451eb0/tests/cupy_tests/core_tests/test_ndarray_scatter.py#L75).